### PR TITLE
Query multiple tables

### PR DESCRIPTION
**Pull request by souad988**
**Project requirements**

1. Create a table named owners with the following columns:

id: integer (set it as autoincremented PRIMARY KEY)
full_name: string
age: integer

2. Create a table named species with the following columns:

id: integer (set it as autoincremented PRIMARY KEY)
name: string

3. Modify animals table:

Make sure that id is set as autoincremented PRIMARY KEY
Remove column species
Add column species_id which is a foreign key referencing species table
Add column owner_id which is a foreign key referencing the owners table

4. Insert the following data into the owners table:

Sam Smith 34 years old.
Jennifer Orwell 19 years old.
Bob 45 years old.
Melody Pond 77 years old.
Dean Winchester 14 years old.
Jodie Whittaker 38 years old.

5. Insert the following data into the species table:

Pokemon
Digimon

6. Modify your inserted animals so it includes the species_id value:

If the name ends in "mon" it will be Digimon
All other animals are Pokemon

7. Modify your inserted animals to include owner information (owner_id):

Sam Smith owns Agumon.
Jennifer Orwell owns Gabumon and Pikachu.
Bob owns Devimon and Plantmon.
Melody Pond owns Charmander, Squirtle, and Blossom.
Dean Winchester owns Angemon and Boarmon.

8. Write queries (using JOIN) to answer the following questions:

What animals belong to Melody Pond?
List of all animals that are pokemon (their type is Pokemon).
List all owners and their animals, remember to include those that don't own any animal.
How many animals are there per species?
List all Digimon owned by Jennifer Orwell.
List all animals owned by Dean Winchester that haven't tried to escape.
Who owns the most animals?

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -9,4 +9,23 @@ CREATE TABLE animals(
    weight_kg         REAL
 );
 
-ALTER TABLE animals ADD species CHAR(100);
\ No newline at end of file
+ALTER TABLE animals ADD species CHAR(100);
+
+CREATE TABLE owners(
+   id  SERIAL PRIMARY KEY     NOT NULL,
```
Comment (arslanbisharat) [inline schema.sql]: - [x] id should be an int

File: schema.sql [hunk 2]
```diff
@@ -9,4 +9,23 @@ CREATE TABLE animals(
...
+   full_name           TEXT    NOT NULL,
+   age  NUMERIC
```
Comment (arslanbisharat) [inline schema.sql]: - [x] age should be an int

File: schema.sql [hunk 3]
```diff
@@ -9,4 +9,23 @@ CREATE TABLE animals(
    weight_kg         REAL
 );
 
-ALTER TABLE animals ADD species CHAR(100);
\ No newline at end of file
+ALTER TABLE animals ADD species CHAR(100);
+
+CREATE TABLE owners(
+   id  SERIAL PRIMARY KEY     NOT NULL,
+   full_name           TEXT    NOT NULL,
```
Comment (arslanbisharat) [inline schema.sql]: - [x] [Optional] full_name should be a string (TEXT should work but string recommended)

File: schema.sql [hunk 4]
```diff
@@ -9,4 +9,23 @@ CREATE TABLE animals(
...
+   age  NUMERIC
+);
+
```
Comment (arslanbisharat) [inline schema.sql:21]: - [x] Create a table named species with the following columns:

```
id: integer (set it as autoincremented PRIMARY KEY)
name: string
```

File: schema.sql [hunk 5]
```diff
@@ -9,4 +9,23 @@ CREATE TABLE animals(
    weight_kg         REAL
```
Comment (arslanbisharat) [inline schema.sql]: - [x] ALTER TABLE animals ADD species_id int REFERENCES species(id)

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by nicupop729 — APPROVED** [review]
Hi @souad988, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
 - [X] Correct git flow practice :heavy_check_mark:
 - [X] Good commit messages :heavy_check_mark:
 - [X] Clean codebase :heavy_check_mark:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**